.github/workflows/ci.yml

```diff
@@ -70,7 +70,7 @@ jobs:
       if: matrix.os == 'ubuntu-latest'
 
     - name: cmake (macos)
-      run: cmake -S . -B out -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=out/install
+      run: cmake -S . -B out -G Ninja -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=out/install -DBYN_ENABLE_LTO=ON
```

Comment:
I'd suggest not changing the mac build since it's what we ship to users atm.

(The linux "clang" build sgtm to change to LTO, since we do have a bunch of other non-LTO builds on linux anyhow.)

Reply:
OK yeah, that's sort of what I had in mind for now too.
I think we do want to ship this to users eventually though if it doesn't end up being too much trouble.

Reply:
Agreed, yeah, would be good to ship this. (And switch testing then - but I don't feel strongly about the meantime either.)

Reply:
Speaking of, I noticed that we still force assertions on. Are we close to no longer requiring that?

Reply:
(also our current command line puts a `-O2` and overrides the default `-O3`)

Reply:
I think the status is not changed from https://github.com/WebAssembly/binaryen/pull/3065 - we were thinking of testing with assertions but shipping to users without, which would need more work. LTO is probably a bigger deal though than assertions, from your data.

Reply:
I found it to be fairly evenly split between them, actually. Although those numbers were probably dominated by LLVM rather than Binaryen. For emscripten-releases I think we do want to test with assertions for LLVM as well. Not sure how much difference it makes on the standalone releases.